smithy-model/src/main/java/software/amazon/smithy/model/node/DefaultNodeDeserializers.java

```diff
@@ -373,8 +374,8 @@ private static boolean isBeanOrBuilderSetter(Method method, Class<?> type, Strin
                 return false;
             }
 
-            // Must either return the target class itself (like a builder) or void.
-            if (method.getReturnType() != void.class && method.getReturnType() != type) {
```

Comment:
What about this?
```
            if (!(method.getReturnType() == void.class || method.getReturnType().isAssignableFrom(type))) {
                return false;
            }
```
Isn't as strict as checking that the return type to be exactly the target, but may be ok? Or at least not specific to sourceLocation.

Reply:
Nice! that is a much more tolerable solution